### PR TITLE
Pallet Gas Burn

### PR DESCRIPTION
**Pull request by pandres95**

### CODE REVIEW

File: pallets/gas-tank/src/traits.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+use frame_support::{weights::Weight, Parameter};
+
+/// Handles
+pub trait PrepaidGasHandler {
```
Comment (olanod) [inline pallets/gas-tank/src/traits.rs:4]: I think it can get a bit simpler and more "gassy", that is, less dependent of the substrat-y concepts of the signed extension. Our trait should just care about checking if the account has a membership with a minimum of burnable weight available and then just burning the actual need weight.

What about something like this
```rust
pub trait GasBurner {
    type Gas;
    type AccountId;

    /// Check if account has a minimum of `gas` to consume. 
    /// Returns the gas that would be left after burning the requested amount or `None` if there's not enough left.
    /// When `gas` is not provided it simply returns the available gas.
    fn check_available_gas(who: &Self::AccountId, gas: Option<&Self::Gas>) -> Option<Self::Gas>;

    /// Spend as much `gas` as possible returning what is left in the tank.
    fn burn_gas(who: &Self::AccountId, gas: &Self::Gas) -> Self::Gas;
}
```

A later trait we can define is a `GasRefiller` with the necessary methods to refuel the tank

**Review by jgutierrezre — APPROVED** [review]
lgtm

Jokes aside, the implementation (and the implemented changes based on @olanod's comments) look sound to me. A good starting point for me to brush up on extensions, as these would be invaluable when implementing feeless txs.

**Comment by pandres95** [comment]
Added base tests.

(Almost) ready to review. Needs adding additional tests to assert some edge cases and add coverage.

**Review by olanod — APPROVED** [review]

**Comment by pandres95** [comment]
@olanod @jgutierrezre final review.

Tests are ready, and `GasBurner` is exported into a different package called `fc-traits-gas-tank`.

**Comment by jgutierrezre** [comment]
Why did you find moving GasBurner into its own pallet necessary?

**Comment by pandres95** [comment]
Yup. And useful, too. Those traits would be implemented by several pallets (i.e. `pallet-communities`, `pallet-nfts` —via `non_fungibles` traits—, `pallet-pass`) that don't really need to know the actual usage of the trait (i.e. the application of the `ChargeTransaction` extension).

**Review by jgutierrezre — APPROVED** [review]
